samples/app-ReporterPlus/ReporterPlus/Bots/ReporterPlusBot.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveCards;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using ReporterPlus.Models;
+using ReporterPlus.Helpers;
+
+namespace ReporterPlus.Bots
+{
+    public class ReporterPlusBot : TeamsActivityHandler
+    {
+        public ReporterPlusBot(IConfiguration configuration) : base()
+        {
+            Constants.MicrosoftAppId = configuration["MicrosoftAppId"];
+            Constants.MicrosoftAppPassword = configuration["MicrosoftAppPassword"];
+            Constants.TenantId = configuration["TenantId"];
+            Constants.BaseUrl = configuration["BaseUrl"];
+            Constants.ServiceUrl = configuration["ServiceUrl"];
+            Constants.OriginatorId = configuration["OriginatorId"];
+            Constants.SenderEmail = configuration["SenderEmail"];
+            Constants.BlobContainerName = configuration["BlobContainerName"];
+            Constants.BlobConnectionString = configuration["BlobConnectionString"];
+        }
+
+        protected override async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name == "composeExtension/fetchTask")
+            {
+                MessagingExtensionAction action = new MessagingExtensionAction();
+                var actionJson = JsonConvert.DeserializeObject<MessagingExtensionActionDeserializer>(turnContext.Activity.Value.ToString());
+                action.CommandId = actionJson.commandId;
+                var task = await this.OnTeamsMessagingExtensionFetchTaskAsync(turnContext, action, cancellationToken);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "composeExtension/submitAction")
+            {
+                MessagingExtensionAction action = new MessagingExtensionAction();
+                var task = await this.OnTeamsMessagingExtensionSubmitActionAsync(turnContext, action, cancellationToken);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "task/fetch")
+            {
+                var actionJson = JsonConvert.DeserializeObject<ActionBase>(turnContext.Activity.Value.ToString());
+                var task = await ReturnViewDetails(actionJson.data.reqId);
+                return CreateInvokeResponse(task);
+            }
+            else if (turnContext.Activity.Name == "adaptiveCard/action")
```

Comment:
I would suggest to add string variables for all such constants used in the file. eg: `string adaptiveCardAction = "adaptiveCard/action";`

Then you could do all string comparisons such as below since we want to compare the contents of the objects and not the references
`if (adaptiveCardAction.Equals(turnContext.Activity.Name))`